pyccel/codegen/utilities.py

```diff
@@ -110,7 +115,14 @@ def compile_fortran(filename, compiler, flags,
     m_code = ' '.join('{}.o'.format(m) for m in modules)
     libs_flags = ' '.join('-l{}'.format(i) for i in libs)
 
-    cmd = '{0} {1} {2} "{3}" {4} "{5}" {6} {7} {8}'.format( \
+    filename = '"{}"'.format(filename)  # in case of spaces in path
+    binary = '"{}"'.format(binary)
+
+    if sys.platform == "win32" and compiler == "mpif90":
+        compiler = "gfortran"
```

Comment:
Why do we change the selected compiler? If this is a default then shouldn't the switch be done when the compiler is selected? If not then shouldn't we use the compiler that the user asked for?

Reply:
The problem here is that `mpif90` doesn't exist on Windows. One rather writes `gfortran` with some command line argument telling it to link MPI libraries. This solution here is not very clean, as it doesn't distinguish whether the user explicitly wants `mpif90`, and whether MPI default compiler was selected. This should indeed be changed.